src/Serializer/Filter/PropertyFilter.php

```diff
@@ -129,18 +134,42 @@ private function getProperties(array $properties, array $whitelist = null): arra
 
         foreach ($properties as $key => $value) {
             if (is_numeric($key)) {
-                if (\in_array($value, $whitelist, true)) {
-                    $result[] = $value;
+                if (\in_array($propertyName = $this->denormalizePropertyName($value), $whitelist, true)) {
+                    $result[] = $propertyName;
                 }
 
                 continue;
             }
 
             if (\is_array($value) && isset($whitelist[$key]) && $recursiveResult = $this->getProperties($value, $whitelist[$key])) {
-                $result[$key] = $recursiveResult;
+                $result[$this->denormalizePropertyName($key)] = $recursiveResult;
+            }
+        }
+
+        return $result;
+    }
+
+    private function denormalizeProperties(array $properties): array
+    {
+        if (null === $this->nameConverter || !$properties) {
+            return $properties;
+        }
+
+        $result = [];
+        foreach ($properties as $key => $value) {
+            if (is_numeric($key)) {
+                $result[] = $this->denormalizePropertyName($value);
+                continue;
```

Comment:
Why a condition for this? should be fine if it's through the same path as below no?

Reply:
the signature of the method expects a string `private function denormalizePropertyName(string $property): string` and I wanted to keep as close as possible of `getProperties()` so it remains unchainged when those values are merge using `array_merge_recursive` or the when the diff with he whitelist is done

Reply:
fair enough! Although, the name converter signature isn't typed so `denormalizePropertyName` could also accept anything